### PR TITLE
Warn instead of erroring when sub is nil

### PR DESCRIPTION
**Pull request by gdbranco**

### CODE REVIEW

**Comment by vkareh** [comment]
But do subscriptions get cleaned up? I thought AMS just marked them as 
`Deprovisioned` (soft delete) so that we can always find them. Has this 
changed recently?

File: cmd/dlt/oidcprovider/cmd.go [hunk 1]
```diff
@@ -87,11 +87,12 @@ func run(cmd *cobra.Command, argv []string) {
 			r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
 			os.Exit(1)
 		} else if sub == nil {
-			r.Reporter.Errorf("Failed to get cluster '%s': %v", r.ClusterKey, err)
-			os.Exit(1)
+			r.Reporter.Warnf("Failed to get subscription for cluster '%s': %v", clusterKey, err)
+			r.Reporter.Warnf("The subscription resource might have already been cleaned, "+
```
Comment (ryankwilliams) [inline cmd/dlt/oidcprovider/cmd.go:91]:
```suggestion
			r.Reporter.Warnf("The subscription resource might have already been cleaned up, "+
```

File: cmd/dlt/operatorrole/cmd.go [hunk 2]
```diff
@@ -117,8 +117,9 @@ func run(cmd *cobra.Command, argv []string) {
 			r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
 			os.Exit(1)
 		} else if sub == nil {
-			r.Reporter.Errorf("Failed to get cluster '%s': %v", r.ClusterKey, err)
-			os.Exit(1)
+			r.Reporter.Warnf("Failed to get subscription for cluster '%s': %v", clusterKey, err)
+			r.Reporter.Warnf("The subscription resource might have already been cleaned, "+
```
Comment (ryankwilliams) [inline cmd/dlt/operatorrole/cmd.go:121]:
```suggestion
			r.Reporter.Warnf("The subscription resource might have already been cleaned up, "+
```

**Comment by vkareh** [comment]
@gdbranco - can you squash locally? Using the tide label messes up the commit messages.

**Pull request closed without merging**